File: 3637-trionic-array-i/3637-trionic-array-i.py
class Solution:
    def isTrionic(self, nums):
        n = len(nums)
        if n < 4:
            return False

        incL =[False] * n
        incR =[False] * n
        incL[0] = True
        for i in range(1,n):
            incL[i] = incL[i-1] and nums[i-1]<nums[i]
        incR[n-1]=True
        for i in range( n-2, -1,-1):
                incR[i] =incR[i+1] and nums[i]< nums[i+1]
            
        for p in range(1, n-2):
                if not incL[p] or nums[p]<= nums[p+1]:
                    continue
                
                q = p+1

                while q < n-1 and nums[q-1] > nums[q]:

                    if incR[q]:
                        return True

                    q += 1
        return False
        

File: 3637-trionic-array-i/3637-trionic-array-i.cpp
class Solution {
public:
    bool isTrionic(vector<int>& nums) {
        int n = nums.size();
        for(int p =1; p <= n-3;p++){
            for(int q =p +1;q<=n-2;q++){
                bool ok = true;
                for(int i =0; i< p;i++){
                    if(nums[i]>=nums[i+1]){
                        ok =false;
                        break;
                    }
                }
                if(!ok) continue;

                for(int i = p; i<q;i++){
                    if(nums[i]<=nums[i+1]){
                        ok = false;
                        break;
                    }
                }
                if(!ok ) continue;
                for(int i =q; i< n-1; i++){
                    if(nums[i]>= nums[i+1]){
                        ok = false;
                        break;
                    }
                }
                if(!ok) continue;
                return true;
            }
        }
        return false;

        
    }
};